internal/registry/ip/ipaddressallocator/ipaddressallocator.go

```diff
@@ -34,38 +34,51 @@ var (
 type Allocator struct {
 	family corev1.IPFamily
 
-	prefix  netip.Prefix
-	firstIP netip.Addr
-	lastIP  netip.Addr
-	size    int64
+	prefixes []netip.Prefix
+	firstIPs []netip.Addr
+	lastIPs  []netip.Addr
+	sizes    []int64
```

Comment:
I think this approach is kind of brittle because you are relying on actual order. As you are using `append()` the order is the same, however from my point of view it makes sense to introduce a struct which groups those fields together.

E.g.

```
type prefixMetaInformation struct {
    prefix  netip.Prefix
    firstIP netip.Addr
    lastIP netip.Addr
    size    int64
}
```

And then consecutively

```
type Allocator struct {
    family corev1.IPFamily

    prefixMetaInformation []prefixMetaInformation
}
```

Reply:
Done, incorporated the above suggested change